### DISCUSSION TITLE
The yellow 'column selected' colour in DataTable

### DISCUSSION THREAD

**Discussion by leemthai** [Textualize/textual#4839] [Q&A]
Hey Guys,

I've got a pretty standard DataTable. The only non-standard bit is  the horizontal bar graph I do with brailled characters. It all works pretty well I think:
![highlight numeric column and select numeric column](https://github.com/user-attachments/assets/a3809fdd-b9c8-481e-b060-a3876c4d7201)

Even when the bar graph column is highlighted, the colors combine pretty well (is the brown highlight colour done semi-transparently?)
![highlight graphic column and select numeric column](https://github.com/user-attachments/assets/27ababe4-614b-4fb8-aac7-7b7a1a421169)

So that's great. The place where it goes wrong (for my eyes/monitor) is if the user selects  a bar graph column:
![highlight numeric column and select graphic column](https://github.com/user-attachments/assets/5d035202-fabb-4d91-b17f-0d7ac63eb49b)
The bright yellow of the cursor selection combined with the bright green of the bar graphs is a pretty yucky combo.

I have tried different colours but nothing really works that well. If I tone the bar graph colours down so they work against the yellow, they look lacklustre/low-contrast in their normal status (against a black background.)

Do you have any ideas? I'm a bit stuck rn. The only thought I had would be maybe an option in DataTable to turn the brightness of the yellow "selection" colour down. Make it more opaque maybe (like the brown highlighted colour is).

Cheers,

**Comment by TomJGooding** [comment]
The 'selected' color in the DataTable is the cursor (try navigating the table with your keyboard). You can target this cursor in CSS with `.datatable--cursor`.

https://textual.textualize.io/widgets/data_table/#component-classes
https://textual.textualize.io/guide/widgets/#component-classes

**Comment by leemthai** [comment]
That's perfect. cheers.
